code/modules/shuttle/shuttle_projector.dm

```diff
@@ -0,0 +1,165 @@
+		var/a_rotation = dir2angle(direction_shuttle_will_move) - dir2angle(direction_shuttle_will_dock)
+		var/b_rotation = dir2angle(direction_shuttle_will_dock) - dir2angle(direction_shuttle_will_move)
+
+		rotate_degrees = abs(b_rotation) > abs(a_rotation) ? a_rotation : b_rotation
+
+		var/rotate_degrees_actual = rotate_degrees
+		if (abs(rotate_degrees_actual) == 180)
+			// BYOND DON'T FLIP SO GOOD
+			// I'M SORRY MY FELLOW OCD GAMERS BUT 180 DOES NOT FUCKING WORK
+			rotate_degrees_actual /= 180
+			move_transform.Turn(rotate_degrees_actual)
+			rotate_degrees_actual *= 179
+
+		// little weird but, we need to "center" the vis_contents before rotating so it works properly instead of going around the shuttle_projector obj
+		undock_transform.Translate(-rotate_width_factor, -rotate_height_factor)
+		undock_transform.Turn(rotate_degrees_actual)
+		undock_transform.Translate(rotate_width_factor, rotate_height_factor)
+
+		// apply the reverse to the docked_transform because the shuttle is already turned while we project it
+		docked_transform.Translate(-rotate_width_factor, -rotate_height_factor)
+		docked_transform.Turn(-rotate_degrees_actual)
+		docked_transform.Translate(rotate_width_factor, rotate_height_factor)
+
+		dock_animation_time += (abs(rotate_degrees_actual) == 180 ? 2 : 1) SECONDS
+
+	// stay centered after scaling/turning
+	undock_transform.Translate(port_width * translate_factor, port_height * translate_factor)
+
+	if (!total_animate_time)
+		total_animate_time = 10 SECONDS
+
+	var/move_animation_time = total_animate_time - dock_animation_time
+
+	// Get the bottom left turf of the bounding box where the shuttle will dock in the stationary port. Origin point of all transforms
+	// Hack hacks hacks! Shuttles are hard okay?
+	var/minx = INFINITY
+	var/miny = INFINITY
+	for (var/I in shuttle_port.ripple_area(stationary_port))
+		var/turf/T = I
+		minx = min(T.x, minx)
+		miny = min(T.y, miny)
+
+	bottom_left = locate(minx, miny, stationary_port.z)
+
+	undock_transform = docked_transform * undock_transform
+	var/matrix/combined_transform = undock_transform * move_transform
+
+	if (inbound)
+		// make sure we turn correctly while docked
+		// start at the end position
+		transform = combined_transform
+		alpha = 0
+	else
+		transform = docked_transform
+
+	vis_contents = projected_turfs
+	forceMove(bottom_left)
+
+	if (inbound)
+		animate(src, transform = undock_transform, easing = CIRCULAR_EASING | EASE_OUT, alpha = docking_alpha, time = move_animation_time)
+		animate(transform = docked_transform, alpha = 255, time = dock_animation_time)
+	else
+		animate(src, transform = undock_transform, alpha = docking_alpha, time = dock_animation_time)
+		animate(transform = combined_transform, easing = CIRCULAR_EASING | EASE_IN, alpha = 0, time = move_animation_time)
+
+	if(!inbound)
+		// rely on remove_ripples to delete us otherwise
+		addtimer(CALLBACK(GLOBAL_PROC, /proc/qdel, src), total_animate_time, TIMER_CLIENT_TIME)
+
```

Comment:
```suggestion
		addtimer(CALLBACK(src, .proc/on_initialization_end), total_animate_time, TIMER_CLIENT_TIME)


/// Handles the aftermath of initializing, after all the deeds are done.
/obj/effect/abstract/shuttle_projector/proc/on_initialization_end()
	qdel(src)
```
This could be cargo culting as I don't remember the exact reasons, but this addtimer pattern has been tried before and found issues with. Probably it being called even is the source is deleted, causing a qdel call to a qdeleted object, which shouldn't happen like this.